server/src/main/java/io/crate/execution/engine/pipeline/ProjectionToProjectorVisitor.java

```diff
@@ -555,6 +556,16 @@ public Projector visitUpdateProjection(final UpdateProjection projection, Contex
         Context context, UpdateProjection projection,
         Collector<ShardResponse, A, Iterable<Row>> collector) {
 
+        // Get Stats to improve ram estimation for the update items
+        assert shardId != null : "ShardId must be provided for updates";
+
+        String indexName = shardId.getIndexName();
+        RelationName relationName = RelationName.fromIndexName(indexName);
```

Comment:
This is somewhat incompatible with the plans for https://github.com/crate/crate/issues/17518

(Or if we ever support updates based on joins for that matter)

I think instead we have to add a `fullDocSizeEstimate()` or something like that to the `UpdateProjection` and already set it in the analyzer/planner where the info of the target relation is fully available.

Or alternatively we'll need a way to lookup the current Relation name for a given indexName/id.

Reply:
Okay, it makes sense to me. Once I am done with my current task, I can follow up with this for 6.0.